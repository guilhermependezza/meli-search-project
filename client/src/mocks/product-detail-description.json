{
    "text": "",
    "plain_text": "1 DE MAYO HOGAR\r\n\r\nCOD: 18169\r\n\r\nSILLITA NIÑO CELESTE GEMPLAST\r\n\r\n\r\nFORMAS DE PAGO: Hasta 12 cuotas sin interés a través de Mercadopago\r\n- Hacemos factura A o B\r\n\r\nFORMAS DE ENTREGA: Envío a través de Mercadoenvios o con flete o transporte\r\n- Consulte valor de envío\r\n- Ofrecemos retiro de nuestra sucursal ubicada en Loma Hermosa, Partido Tres de Febrero (a 1 cuadra de Ruta 8)\r\n\r\nIMPORTANTE:\r\n- Consulte stock antes de ofertar\r\n- Aceptamos mercadopago\r\n- Horarios de Atención de Lunes a Sábados de 9 a 13 y de 16 a 20.30 Hs.\r\n- Local a la calle con salón de exposición\r\n- Haga todas las preguntas necesarias antes de ofertar su consulta no molesta\r\n- En caso de necesitar factura A informar luego de realizada la compra. De lo contrario se emitirá factura B.\r\n\r\n\r\nPuede ver todos nuestros productos publicados en : http://eshops.mercadolibre.com.ar/1demayo+virtual",
    "last_updated": "2017-09-06T21:42:32.000Z",
    "date_created": "2017-01-27T15:29:45.000Z",
    "snapshot": {
        "url": "http://descriptions.mlstatic.com/D-MLA651772871.jpg?hash=7C1921A0BA7F114C7011A26CBFD3B73_938x163",
        "width": 938,
        "height": 163,
        "status": "OK"
    }
}