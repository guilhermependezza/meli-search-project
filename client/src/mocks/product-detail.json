{
    "author": {
        "name": "Guilherme",
        "lastname": "Sousa"
    },
    "item": {
        "id": "MLA651772871",
        "site_id": "MLA",
        "title": "Silla Plastica  Gemplast Niños",
        "subtitle": null,
        "seller_id": 142578345,
        "category_id": "MLA74619",
        "official_store_id": null,
        "price": 159,
        "base_price": 159,
        "original_price": null,
        "currency_id": "ARS",
        "initial_quantity": 20,
        "available_quantity": 1,
        "sold_quantity": 5,
        "sale_terms": [],
        "buying_mode": "buy_it_now",
        "listing_type_id": "gold_special",
        "start_time": "2017-01-27T15:29:44.000Z",
        "stop_time": "2037-01-22T15:29:44.000Z",
        "condition": "new",
        "permalink": "https://articulo.mercadolibre.com.ar/MLA-651772871-silla-plastica-gemplast-ninos-_JM",
        "thumbnail": "http://mla-s2-p.mlstatic.com/887660-MLA31042543025_062019-I.jpg",
        "secure_thumbnail": "https://mla-s2-p.mlstatic.com/887660-MLA31042543025_062019-I.jpg",
        "pictures": [
            {
                "id": "887660-MLA31042543025_062019",
                "url": "http://mla-s2-p.mlstatic.com/887660-MLA31042543025_062019-O.jpg",
                "secure_url": "https://mla-s2-p.mlstatic.com/887660-MLA31042543025_062019-O.jpg",
                "size": "228x292",
                "max_size": "228x292",
                "quality": ""
            },
            {
                "id": "874054-MLA31042482913_062019",
                "url": "http://mla-s2-p.mlstatic.com/874054-MLA31042482913_062019-O.jpg",
                "secure_url": "https://mla-s2-p.mlstatic.com/874054-MLA31042482913_062019-O.jpg",
                "size": "466x500",
                "max_size": "800x857",
                "quality": ""
            },
            {
                "id": "995686-MLA26119091951_102017",
                "url": "http://mla-s2-p.mlstatic.com/995686-MLA26119091951_102017-O.jpg",
                "secure_url": "https://mla-s2-p.mlstatic.com/995686-MLA26119091951_102017-O.jpg",
                "size": "500x489",
                "max_size": "800x783",
                "quality": ""
            },
            {
                "id": "949776-MLA26119102650_102017",
                "url": "http://mla-s2-p.mlstatic.com/949776-MLA26119102650_102017-O.jpg",
                "secure_url": "https://mla-s2-p.mlstatic.com/949776-MLA26119102650_102017-O.jpg",
                "size": "500x492",
                "max_size": "800x788",
                "quality": ""
            },
            {
                "id": "932062-MLA31042482895_062019",
                "url": "http://mla-s2-p.mlstatic.com/932062-MLA31042482895_062019-O.jpg",
                "secure_url": "https://mla-s2-p.mlstatic.com/932062-MLA31042482895_062019-O.jpg",
                "size": "464x500",
                "max_size": "800x861",
                "quality": ""
            }
        ],
        "video_id": null,
        "descriptions": [
            {
                "id": "MLA651772871-1264068382"
            }
        ],
        "accepts_mercadopago": true,
        "non_mercado_pago_payment_methods": [],
        "shipping": {
            "mode": "me2",
            "methods": [],
            "tags": [],
            "dimensions": null,
            "local_pick_up": true,
            "free_shipping": false,
            "logistic_type": "drop_off",
            "store_pick_up": false
        },
        "international_delivery_mode": "none",
        "seller_address": {
            "city": {
                "name": "Loma Hermosa"
            },
            "state": {
                "id": "AR-B",
                "name": "Buenos Aires"
            },
            "country": {
                "id": "AR",
                "name": "Argentina"
            },
            "search_location": {
                "neighborhood": {
                    "id": "TUxBQkxPTTI0NTg1MA",
                    "name": "Loma Hermosa"
                },
                "city": {
                    "id": "TUxBQ1RSRTMxODE5NA",
                    "name": "Tres de febrero"
                },
                "state": {
                    "id": "TUxBUEdSQWVmNTVm",
                    "name": "Bs.As. G.B.A. Oeste"
                }
            },
            "latitude": -34.56765,
            "longitude": -58.613792,
            "id": 107047667
        },
        "seller_contact": null,
        "location": {},
        "geolocation": {
            "latitude": -34.56765,
            "longitude": -58.613792
        },
        "coverage_areas": [],
        "attributes": [
            {
                "id": "BRAND",
                "name": "Marca",
                "value_id": "4219213",
                "value_name": "Gemplast",
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "CHAIR_TYPE",
                "name": "Tipo de silla",
                "value_id": null,
                "value_name": "Niño",
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "DEPTH",
                "name": "Profundidad",
                "value_id": "-1",
                "value_name": null,
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "HEIGHT",
                "name": "Altura",
                "value_id": null,
                "value_name": "50 cm",
                "value_struct": {
                    "number": 50,
                    "unit": "cm"
                },
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "IS_FOLDABLE",
                "name": "Es plegable",
                "value_id": "242084",
                "value_name": "No",
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "ITEM_CONDITION",
                "name": "Condición del ítem",
                "value_id": "2230284",
                "value_name": "Nuevo",
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "LEGS_MATERIAL",
                "name": "Material de las patas",
                "value_id": "973765",
                "value_name": "Plástico",
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "LENGTH",
                "name": "Largo",
                "value_id": null,
                "value_name": "28 cm",
                "value_struct": {
                    "number": 28,
                    "unit": "cm"
                },
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "MANUFACTURER",
                "name": "Fabricante",
                "value_id": null,
                "value_name": "Gemplast",
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "MODEL",
                "name": "Modelo",
                "value_id": null,
                "value_name": "Niños",
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "STRUCTURE_MATERIAL",
                "name": "Material de la estructura",
                "value_id": "2748302",
                "value_name": "Plástico",
                "value_struct": null,
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "TOTAL_HEIGHT",
                "name": "Altura total",
                "value_id": null,
                "value_name": "50 cm",
                "value_struct": {
                    "number": 50,
                    "unit": "cm"
                },
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            },
            {
                "id": "WIDTH",
                "name": "Ancho",
                "value_id": null,
                "value_name": "35 cm",
                "value_struct": {
                    "number": 35,
                    "unit": "cm"
                },
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            }
        ],
        "warnings": [],
        "listing_source": "",
        "variations": [
            {
                "id": 30765330672,
                "price": 159,
                "attribute_combinations": [
                    {
                        "id": "COLOR",
                        "name": "Color",
                        "value_id": "52021",
                        "value_name": "Celeste",
                        "value_struct": null
                    },
                    {
                        "id": "PATTERN_NAME",
                        "name": "Nombre del diseño",
                        "value_id": null,
                        "value_name": "Undefined",
                        "value_struct": null
                    }
                ],
                "available_quantity": 1,
                "sold_quantity": 5,
                "sale_terms": [],
                "picture_ids": [
                    "887660-MLA31042543025_062019",
                    "874054-MLA31042482913_062019",
                    "995686-MLA26119091951_102017",
                    "949776-MLA26119102650_102017",
                    "932062-MLA31042482895_062019"
                ],
                "catalog_product_id": null
            }
        ],
        "status": "active",
        "sub_status": [],
        "tags": [
            "poor_quality_picture",
            "immediate_payment",
            "cart_eligible"
        ],
        "warranty": null,
        "catalog_product_id": null,
        "domain_id": "MLA-GARDEN_CHAIRS",
        "parent_item_id": null,
        "differential_pricing": null,
        "deal_ids": [],
        "automatic_relist": false,
        "date_created": "2017-01-27T15:29:45.000Z",
        "last_updated": "2019-06-18T12:23:42.000Z",
        "health": 0.6
    }
}