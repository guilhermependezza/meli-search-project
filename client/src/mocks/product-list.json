{  
    "author":{  
       "name":"Guilherme",
       "lastname":"Pendezza"
    },
    "items":[  
       {  
          "id":"MLA772565193",
          "site_id":"MLA",
          "title":"Por Que A Vov Guardou A Calcinha Na Geladeira? : Max Wallac",
          "seller":{  
             "id":10119554,
             "power_seller_status":"platinum",
             "car_dealer":false,
             "real_estate_agency":false,
             "tags":[  
 
             ]
          },
          "price":1661,
          "currency_id":"ARS",
          "available_quantity":1,
          "sold_quantity":0,
          "buying_mode":"buy_it_now",
          "listing_type_id":"gold_special",
          "stop_time":"2039-02-18T04:00:00.000Z",
          "condition":"new",
          "permalink":"https://articulo.mercadolibre.com.ar/MLA-772565193-por-que-a-vov-guardou-a-calcinha-na-geladeira-max-wallac-_JM",
          "thumbnail":"http://mla-s2-p.mlstatic.com/947591-MLA29489378962_022019-I.jpg",
          "accepts_mercadopago":true,
          "installments":{  
             "quantity":12,
             "amount":226.68,
             "rate":63.77,
             "currency_id":"ARS"
          },
          "address":{  
             "state_id":"AR-C",
             "state_name":"Capital Federal",
             "city_id":"",
             "city_name":"Barrio Norte"
          },
          "shipping":{  
             "free_shipping":false,
             "mode":"me2",
             "tags":[  
 
             ],
             "logistic_type":"cross_docking",
             "store_pick_up":false
          },
          "seller_address":{  
             "id":"",
             "comment":"",
             "address_line":"",
             "zip_code":"",
             "country":{  
                "id":"AR",
                "name":"Argentina"
             },
             "state":{  
                "id":"AR-C",
                "name":"Capital Federal"
             },
             "city":{  
                "id":"",
                "name":"Barrio Norte"
             },
             "latitude":"",
             "longitude":""
          },
          "attributes":[  
             {  
                "name":"Condición del ítem",
                "value_id":"2230284",
                "value_name":"Nuevo",
                "value_struct":null,
                "attribute_group_id":"",
                "attribute_group_name":"",
                "source":1,
                "id":"ITEM_CONDITION"
             }
          ],
          "original_price":null,
          "category_id":"MLA1227",
          "official_store_id":null,
          "catalog_product_id":null,
          "tags":[  
             "poor_quality_picture",
             "immediate_payment",
             "cart_eligible"
          ]
       },
       {  
          "id":"MLA675084756",
          "site_id":"MLA",
          "title":"Abra O Coracao E Feche A Geladeira De Terkeurst Lysa",
          "seller":{  
             "id":247423156,
             "power_seller_status":null,
             "car_dealer":false,
             "real_estate_agency":false,
             "tags":[  
 
             ]
          },
          "price":1906.35,
          "currency_id":"ARS",
          "available_quantity":1,
          "sold_quantity":0,
          "buying_mode":"buy_it_now",
          "listing_type_id":"gold_special",
          "stop_time":"2037-07-14T10:59:00.000Z",
          "condition":"new",
          "permalink":"https://articulo.mercadolibre.com.ar/MLA-675084756-abra-o-coracao-e-feche-a-geladeira-de-terkeurst-lysa-_JM",
          "thumbnail":"http://mla-s1-p.mlstatic.com/817189-MLA25767059581_072017-I.jpg",
          "accepts_mercadopago":true,
          "installments":{  
             "quantity":12,
             "amount":260.17,
             "rate":63.77,
             "currency_id":"ARS"
          },
          "address":{  
             "state_id":"AR-C",
             "state_name":"Capital Federal",
             "city_id":"TUxBQlZJTDE2MDBa",
             "city_name":"Villa Pueyrredón"
          },
          "shipping":{  
             "free_shipping":true,
             "mode":"me2",
             "tags":[  
 
             ],
             "logistic_type":"xd_drop_off",
             "store_pick_up":false
          },
          "seller_address":{  
             "id":"",
             "comment":"",
             "address_line":"",
             "zip_code":"",
             "country":{  
                "id":"AR",
                "name":"Argentina"
             },
             "state":{  
                "id":"AR-C",
                "name":"Capital Federal"
             },
             "city":{  
                "id":"TUxBQlZJTDE2MDBa",
                "name":"Villa Pueyrredón"
             },
             "latitude":"",
             "longitude":""
          },
          "attributes":[  
             {  
                "id":"ITEM_CONDITION",
                "name":"Condición del ítem",
                "value_id":"2230284",
                "value_name":"Nuevo",
                "value_struct":null,
                "attribute_group_id":"OTHERS",
                "attribute_group_name":"Otros",
                "source":1
             }
          ],
          "original_price":null,
          "category_id":"MLA40622",
          "official_store_id":null,
          "catalog_product_id":null,
          "tags":[  
             "poor_quality_picture",
             "immediate_payment",
             "cart_eligible"
          ]
       },
       {  
          "id":"MLA678733059",
          "site_id":"MLA",
          "title":"Que Tem Na Geladeira? O De Lobo Rita",
          "seller":{  
             "id":247423156,
             "power_seller_status":null,
             "car_dealer":false,
             "real_estate_agency":false,
             "tags":[  
 
             ]
          },
          "price":2521.95,
          "currency_id":"ARS",
          "available_quantity":1,
          "sold_quantity":0,
          "buying_mode":"buy_it_now",
          "listing_type_id":"gold_special",
          "stop_time":"2037-08-09T23:19:26.000Z",
          "condition":"new",
          "permalink":"https://articulo.mercadolibre.com.ar/MLA-678733059-que-tem-na-geladeira-o-de-lobo-rita-_JM",
          "thumbnail":"http://mla-s2-p.mlstatic.com/875778-MLA25871467132_082017-I.jpg",
          "accepts_mercadopago":true,
          "installments":{  
             "quantity":12,
             "amount":344.18,
             "rate":63.77,
             "currency_id":"ARS"
          },
          "address":{  
             "state_id":"AR-C",
             "state_name":"Capital Federal",
             "city_id":"TUxBQlZJTDE2MDBa",
             "city_name":"Villa Pueyrredón"
          },
          "shipping":{  
             "free_shipping":true,
             "mode":"me2",
             "tags":[  
                "mandatory_free_shipping"
             ],
             "logistic_type":"xd_drop_off",
             "store_pick_up":false
          },
          "seller_address":{  
             "id":"",
             "comment":"",
             "address_line":"",
             "zip_code":"",
             "country":{  
                "id":"AR",
                "name":"Argentina"
             },
             "state":{  
                "id":"AR-C",
                "name":"Capital Federal"
             },
             "city":{  
                "id":"TUxBQlZJTDE2MDBa",
                "name":"Villa Pueyrredón"
             },
             "latitude":"",
             "longitude":""
          },
          "attributes":[  
             {  
                "name":"Condición del ítem",
                "value_id":"2230284",
                "value_name":"Nuevo",
                "value_struct":null,
                "attribute_group_id":"OTHERS",
                "attribute_group_name":"Otros",
                "source":1,
                "id":"ITEM_CONDITION"
             }
          ],
          "original_price":null,
          "category_id":"MLA40622",
          "official_store_id":null,
          "catalog_product_id":null,
          "tags":[  
             "poor_quality_picture",
             "immediate_payment",
             "cart_eligible"
          ]
       },
       {  
          "id":"MLA679648904",
          "site_id":"MLA",
          "title":"O Pinguim De Geladeira A Preguiça E A Energia Elétrica De Sé",
          "seller":{  
             "id":247423156,
             "power_seller_status":null,
             "car_dealer":false,
             "real_estate_agency":false,
             "tags":[  
 
             ]
          },
          "price":2051.52,
          "currency_id":"ARS",
          "available_quantity":1,
          "sold_quantity":0,
          "buying_mode":"buy_it_now",
          "listing_type_id":"gold_special",
          "stop_time":"2037-08-16T15:58:29.000Z",
          "condition":"new",
          "permalink":"https://articulo.mercadolibre.com.ar/MLA-679648904-o-pinguim-de-geladeira-a-preguica-e-a-energia-eletrica-de-se-_JM",
          "thumbnail":"http://mla-s1-p.mlstatic.com/723540-MLA25895346432_082017-I.jpg",
          "accepts_mercadopago":true,
          "installments":{  
             "quantity":12,
             "amount":279.98,
             "rate":63.77,
             "currency_id":"ARS"
          },
          "address":{  
             "state_id":"AR-C",
             "state_name":"Capital Federal",
             "city_id":"TUxBQlZJTDE2MDBa",
             "city_name":"Villa Pueyrredón"
          },
          "shipping":{  
             "free_shipping":true,
             "mode":"me2",
             "tags":[  
                "mandatory_free_shipping"
             ],
             "logistic_type":"xd_drop_off",
             "store_pick_up":false
          },
          "seller_address":{  
             "id":"",
             "comment":"",
             "address_line":"",
             "zip_code":"",
             "country":{  
                "id":"AR",
                "name":"Argentina"
             },
             "state":{  
                "id":"AR-C",
                "name":"Capital Federal"
             },
             "city":{  
                "id":"TUxBQlZJTDE2MDBa",
                "name":"Villa Pueyrredón"
             },
             "latitude":"",
             "longitude":""
          },
          "attributes":[  
             {  
                "attribute_group_name":"Otros",
                "source":1,
                "id":"ITEM_CONDITION",
                "name":"Condición del ítem",
                "value_id":"2230284",
                "value_name":"Nuevo",
                "value_struct":null,
                "attribute_group_id":"OTHERS"
             }
          ],
          "original_price":null,
          "category_id":"MLA40622",
          "official_store_id":null,
          "catalog_product_id":null,
          "tags":[  
             "poor_quality_picture",
             "immediate_payment",
             "cart_eligible"
          ]
       }
    ]
 }