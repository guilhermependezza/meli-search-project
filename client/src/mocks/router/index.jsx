import React from 'react';
import { BrowserRouter } from 'react-router-dom';

export const mockedRouter = component => <BrowserRouter>{component}</BrowserRouter>;
