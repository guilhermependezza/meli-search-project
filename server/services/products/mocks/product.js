module.exports = {
  productId: 'MLA772565193',
  productTitle: 'Por Que A Vov Guardou A Calcinha Na Geladeira? : Max Wallac',
  descriptionFields: [
    'date_created', 'last_updated', 'plain_text', 'snapshot', 'text'
  ]
};
