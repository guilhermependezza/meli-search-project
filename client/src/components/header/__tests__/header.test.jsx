import React from 'react';
import { create } from 'react-test-renderer';
import { mockedRouter } from 'mocks/router';

// Component
import Header from 'components/header';

describe('Header', () => {
  it('should match snapshot', () => {
    const header = create(mockedRouter(<Header />)).toJSON();
    expect(header).toMatchSnapshot();
  });
});
