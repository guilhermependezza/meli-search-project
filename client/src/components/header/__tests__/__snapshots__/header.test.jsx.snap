// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  className="header"
>
  <div
    className="wrapper"
  >
    <a
      className="logo-link"
      href="/"
      onClick={[Function]}
    />
    <form
      className="searchbar flex-wrapper align-center justify-center"
      onSubmit={[Function]}
    >
      <input
        className="input"
        defaultValue=""
        name="search"
        placeholder="Do que vocÃª precisa?"
        type="text"
      />
      <button
        className="button"
        type="submit"
      >
        <img
          alt="Pesquisar"
          src="/images/ic_Search.png"
        />
      </button>
    </form>
  </div>
</header>
`;
